### ISSUE TITLE
Gnirehtet don't work on android Q.

### ISSUE THREAD

**Issue by JosueBlok** [Genymobile/gnirehtet#230]
is possible run gnirehtet on android Q?

**Comment by rom1v** [comment]
What happens on Android Q? (in the console when you run it, and in `adb logcat`)

**Comment by JosueBlok** [comment]
Gnirehtet run work but the message don't appear in my phone. 

adb logcat returned this

```
10-01 08:42:57.176  1447  1934 D PowerUI : can't show warning due to - plugged: true status unknown: false
10-01 08:42:57.185   920   982 I UsageStatsService: User[0] Flushing usage stats to disk
10-01 08:42:57.259   920   982 E AppIdleHistory: Error writing app idle file for user 0
10-01 08:42:57.259   920   982 I UsageStatsService: User[10] Flushing usage stats to disk
10-01 08:42:57.305   920   982 E AppIdleHistory: Error writing app idle file for user 10
10-01 08:42:58.859 30227 30362 D ClClient: Not sending keepalive.  Current connection state=STOPPED
10-01 08:43:00.010 30227  6225 D EventNotificationJob: Running EventNotificationJob, isDetail=true
10-01 08:43:00.022 30227  6225 D EventNotificationJob:   finished processing events
10-01 08:43:00.044  5398  5553 I com.newrelic.android: Harvester: connected
10-01 08:43:00.044  5398  5553 I com.newrelic.android: Harvester: Sending 0 HTTP transactions.
10-01 08:43:00.044  5398  5553 I com.newrelic.android: Harvester: Sending 0 HTTP errors.
10-01 08:43:00.044  5398  5553 I com.newrelic.android: Harvester: Sending 0 activity traces.
10-01 08:43:00.044  5398  5553 I com.newrelic.android: Harvester: Sending 0 analytics events.
10-01 08:43:00.065  5398  5553 E com.newrelic.android: Failed to retrieve collector response: Unable to resolve host "mobile-collector.newrelic.com": No address associated with hostname
10-01 08:43:00.066  5398  5553 E com.newrelic.android: HarvestConnection: Attempting to convert network exception java.net.UnknownHostException to error code. 
```

**Comment by rom1v** [comment]
Could you paste the output when you execute `./gnirehtet run`?

If you get `Broadcast completed: result=0`, you need the last version on `master` (not released yet). See #227.

**Comment by JosueBlok** [comment]
Yeah i get `Broadcast completed: result=0`
```
2019-10-01 09:14:12.535 INFO Main: Starting relay server...
2019-10-01 09:14:12.535 INFO Main: Checking gnirehtet client...
2019-10-01 09:14:12.539 INFO Relay: Relay server started
2019-10-01 09:14:12.649 INFO Main: Starting client...
Broadcasting: Intent { act=com.genymobile.gnirehtet.START flg=0x400000 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver }
Broadcast completed: result=0
```

**Comment by A-Fat-Duck** [comment]
I'm getting this exact same error, just updated to the Android Q open beta on my OnePlus 6T

> Yeah i get `Broadcast completed: result=0`
> 
> ```
> 2019-10-01 09:14:12.535 INFO Main: Starting relay server...
> 2019-10-01 09:14:12.535 INFO Main: Checking gnirehtet client...
> 2019-10-01 09:14:12.539 INFO Relay: Relay server started
> 2019-10-01 09:14:12.649 INFO Main: Starting client...
> Broadcasting: Intent { act=com.genymobile.gnirehtet.START flg=0x400000 cmp=com.genymobile.gnirehtet/.GnirehtetControlReceiver }
> Broadcast completed: result=0
> ```

**Comment by rom1v** [comment]
Yes, this is expected, and fixed on current `master`. https://github.com/Genymobile/gnirehtet/issues/230#issuecomment-537024582

**Comment by imonlygitting** [comment]
can someone tell us if the Android Q issue has been fixed? above response suggest so, but for newbies who don't do compiling is there any way to get a fixed update executable? Many thanks in advance!

**Comment by rom1v** [comment]
Please test this fixed version: https://github.com/Genymobile/gnirehtet/issues/227#issuecomment-553581824

**Comment by AxForest** [comment]
> 
> 
> Please test this fixed version: [#227 (comment)](https://github.com/Genymobile/gnirehtet/issues/227#issuecomment-553581824)

Works for me. 

Pixel3a - Andriod Q

**Comment by rom1v** [comment]
@AxForest Thank you for your test. Since then, v2.4 is out, where the problem is fixed.
